x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import fetch from 'node-fetch';
+import querystring from 'querystring';
+import { Logger } from 'src/core/server';
+import { RequestHandlerContext, KibanaRequest, KibanaResponseFactory } from 'src/core/server';
+import { ConfigType } from '../index';
+
+interface IAppSearchRequestParams<ResponseBody> {
+  config: ConfigType;
+  log: Logger;
+  path: string;
+  hasValidData?: (body?: ResponseBody) => boolean;
+}
+
+/**
+ * This function creates a handler, in a standard way, for handling App Search
+ * API requests.
+ *
+ * This handler assumes that it will essentially just proxy the App Search API
+ * request, so the request body and request parameters, and body are simply
+ * passed through.
+ */
+export function createAppSearchRequestHandler<ResponseBody>({
+  config,
+  log,
+  path,
+  hasValidData = () => true,
+}: IAppSearchRequestParams<ResponseBody>) {
+  return async (
+    _context: RequestHandlerContext,
+    request: KibanaRequest<unknown, Readonly<{}>, unknown>,
+    response: KibanaResponseFactory
+  ) => {
+    try {
+      const enterpriseSearchUrl = config.host as string;
+      const params = querystring.stringify(request.query);
+      const url = `${encodeURI(enterpriseSearchUrl)}${path}?${params}`;
```

Comment:
If it's cool, I'd like us to tweak this a bit so we don't have a trailing `?` for routes that don't actually have params:

```suggestion
      const params = request.query ? `?${querystring.stringify(request.query)}` : '';
      const url = `${encodeURI(enterpriseSearchUrl)}${path}${params}`;
```

Thoughts? We can definitely reformat if the ternary is too annoying to read

Reply:
No no, this is a good change. I meant to circle back and handle that.